### PR TITLE
Migrate to pip-tools and Hatch

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by ofek** [comment]
Have you tried Hatch and/or Hatchling? 🙂 

Note: Hatchling is now the default in the official [Python packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/)

File: pyproject.toml [hunk 1]
```diff
@@ -1,16 +1,18 @@
 [build-system]
-requires = ["poetry-core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
```
Comment (ofek) [inline pyproject.toml:3]:
```suggestion
requires = ["hatchling"]
build-backend = "hatchling.build"
```

File: pyproject.toml [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+[project]
 name = "ruff-lsp"
-version = "0.0.9"
-description = "A Language Server Protocol implementation for Ruff."
-authors = ["Charlie Marsh <charlie.r.marsh@gmail.com>"]
-maintainers = ["Charlie Marsh <charlie.r.marsh@gmail.com>"]
-license = "MIT"
-readme = "README.md"
-repository = "https://github.com/charliermarsh/ruff-lsp"
+dynamic = ["version", "description"]
```
Comment (ofek) [inline pyproject.toml:7]:
```suggestion
dynamic = ["version"]
description = "A Language Server Protocol implementation for Ruff."
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
+authors = [
+    { name = "Charlie Marsh", email = "charlie.r.marsh@gmail.com" },
+]
+maintainers = [
+    { name = "Charlie Marsh", email = "charlie.r.marsh@gmail.com" },
+]
+requires-python = ">=3.7"
+license = { file = "LICENSE" }
```
Comment (ofek) [inline pyproject.toml:15]: https://peps.python.org/pep-0639/

```suggestion
license = "MIT"
```

**Comment by charliermarsh** [comment]
Would it be unusual to use Hatch or Hatchling but do my own environment management? That’s sort of why I played around with Flit here.

File: pyproject.toml [hunk 4]
```diff
@@ -29,20 +31,22 @@ classifiers = [
...
+    "mypy==0.991",
+    "python-lsp-jsonrpc==1.0.0",
+]
 
-[tool.poetry.dev-dependencies]
-black = "==22.12.0"
-mypy = "==0.991"
-python-lsp-jsonrpc = "==1.0.0"
+[project.scripts]
+ruff-lsp = "ruff_lsp.__main__:main"
```
Comment (ofek) [inline pyproject.toml:50]:
```suggestion

[tool.hatch.version]
path = "ruff_lsp/__init__.py"

```

File: ruff_lsp/__init__.py [hunk 5]
```diff
@@ -1,8 +1,3 @@
-import sys
+"""A Language Server Protocol implementation for Ruff."""
```
Comment (ofek) [inline ruff_lsp/__init__.py:2]:
```suggestion
```

**Review by ofek — COMMENTED** [review]
If you want! This also unlocks the use of [plugins](https://hatch.pypa.io/latest/plugins/about/) like [hatch-mypyc](https://github.com/ofek/hatch-mypyc), [hatch-vcs](https://github.com/ofek/hatch-vcs), etc.

**Comment by ofek** [comment]
> Would it be unusual to use Hatch or Hatchling but do my own environment management? That’s sort of why I played around with Flit here.

Nope!

- https://hatch.pypa.io/latest/build/#packaging-ecosystem
- https://hatch.pypa.io/latest/meta/faq/#interoperability

**Comment by charliermarsh** [comment]
Stellar, thank you! I would prefer to use Hatch :)

**Pull request merged**